src/dart/ast.rs

```diff
@@ -385,53 +537,113 @@ impl Type {
     }
 }
 
+/// An expression.
 #[derive(Debug)]
 pub enum Expr {
+    /// Comments that describe the expression.
     Comments(Vec<Span>, Node<Expr>),
+    ///A unary operation (`!x`, `~x`, etc.).
     Unary(UnOp, Node<Expr>),
+    /// A binary operation (`a + b`, `a * b`, etc.).
     Binary(BinOp, Node<Expr>, Node<Expr>),
+    /// A conditional expression (`a ? b : c`).
     Conditional(Node<Expr>, Node<Expr>, Node<Expr>),
+    /// An expression that checks the type (`var is type`).
     Is(Node<Expr>, Node<Type>),
+    /// A negated expression that checks the type (`var !is type`).
     IsNot(Node<Expr>, Node<Type>),
+    /// An expression that acts as a cast (`x as Type`).
     As(Node<Expr>, Node<Type>),
+    /// A sequence of operator, method, getter or setter invocations.
+    ///
+    /// `expression.suffix`
     Suffix(Node<Expr>, Suffix),
+    /// An identifier (`x`).
     Identifier(Symbol),
+    /// A function object that has access to variables in its lexical scope,
+    /// even when the function is used outside of its original scope.
+    ///
+    ///`(a, b) => a + b`
     Closure(FnSig, FnBody),
+    /// A constructor call.
+    ///
+    /// `new Constructor()`
     New {
         const_: bool,
         path: Node<Qualified>,
         args: Args,
     },
+    /// Zero-based ordered group of objects.
+    ///
+    /// `var list = [1, 2, 3]`
     List {
         const_: bool,
         element_ty: Option<Node<Type>>,
         elements: Vec<Node<Expr>>,
     },
+    /// An object that associates keys and values.
+    ///
+    /// `var map = { 'k1' : 'v1', 'k2' : 'v2', 'k3' : 'v3'}`
     Map {
         const_: bool,
         kv_ty: Option<(Node<Type>, Node<Type>)>,
         kv: Vec<(Node<Expr>, Node<Expr>)>,
     },
+    /// A number (`1.2`).
     Number(Symbol),
+    /// Adjacent strings are implicitly concatenated
+    /// to form a single string literal.
```

Comment:
I think "One or more string literals, implicitly concatenated to form a single \`String\`." is a bit more accurate.